src/vs/workbench/services/editor/common/editorResolverService.ts

```diff
@@ -110,7 +110,7 @@ export type DiffEditorInputFactoryFunction = (diffEditorInput: IResourceDiffEdit
 export type MergeEditorInputFactoryFunction = (mergeEditorInput: IResourceMergeEditorInput, group: IEditorGroup) => EditorInputFactoryResult;
 
 export type EditorInputFactoryObject = {
-	createEditorInput: EditorInputFactoryFunction;
+	createEditorInput?: EditorInputFactoryFunction;
```

Comment:
I wonder if we can express a type here that requires one of the four to be defined. I think having them all undefined makes little sense.

Reply:
Apparently the answer is yes https://stackoverflow.com/questions/48230773/how-to-create-a-partial-like-that-requires-a-single-property-to-be-set